rfcs/001-foundational-architecture.md

```diff
@@ -0,0 +1,564 @@
+  "info": "Click the button to continue."
+}
+```
+
+##### Retrieving a message
+
+```jsx
+import translate from "sdk";
+
+console.log(translate("example", { name: "Samuel" }));
+```
+
+##### Output
+
+```
+>> Welcome Samuel to this example.
+```
+
+#### Observations
+
+1. A variety of good and adopted open source SDKs exist [[1](https://github.com/ivanhofer/typesafe-i18n), [2](https://github.com/formatjs/formatjs), [3](https://pub.dev/packages/flutter_i18n), [4](https://github.com/solidjs-community/solid-primitives/tree/main/packages/i18n), ...]. Each serves a different programming language, framework, niche, or feature.
+
+2. The internals are identical:  
+   `Resource` -> `Reference and Format a Message` -> `Output`
+
+#### Proposal
+
+Leverage existing SDKs instead of forcing a specific inlang SDK.
+
+**Pros**
+
+- Existing codebases can adopt inlang.
+- Different SDKs solve different problems.
+
+**Cons**
+
+- The user experience could potentially be higher with a dedicated SDK.
+
+#### Considered alternative
+
+Develop (and require) a dedicated SDK developed by inlang.
+
+**Pros**
+
+- Potentially better user experience by delivering an "end to end" solution.
+
+**Cons**
+
+- Adoption is severely limited.
+- Different SDKs exist because a "one size fits all" solution likely does not work.
+
+### Dev tools
+
+Using an SDK does not relieve developers from two extremely time-consuming and ever-repeating tasks:
+
+1. Extracting hard-coded strings.
+
+   Developers have to manually copy & paste hardcoded strings into resources. That process can take weeks.
+
+2. Validating translations.
+
+   No widely adopted tools exist to validate translations: Does the German translation exist? Is the French translation correct? Does the UI work as expected for Arabic?
+
+Furthermore, the DX of localizing software, in general, is improvable. The pseudocode below illustrates some problems:
+
+```jsx
+import translate from "i18n-sdk";
+import Translated from 'i18n-sdk';
+
+// What is info referring to?
+<p>{translate('info')}</p>
+
+// 1. Ugh ... do I need to wrap every element in a Translated component now?
+// 2. Okay, I see what `info` is referring to now, but
+//    changing the text "Click the button..." will not
+//    be reflected in the resource! I still have to manually
+//    edit the resource file!
+<Translated id="info">
+  <p>Click the button to continue.</p>
+</Translated>
+```
+
+> At this point you might be wondering why the text "Click the button to continue." itself is not used as key/id. The reasons stems from changing the text and thereby losing the connection to the translations or invalidating them. The Fluent project explained their rationale of using ids [here](https://github.com/projectfluent/fluent/wiki/Fluent-vs-gettext#translation-invalidation).
+
+#### Observations
+
+1. Dev tools beyond SDKs are required to make localization effortless.
+
+2. A CLI to extract and validate messages enables CI/CD.
```

Comment:
Why? Not explained in the previous paragraphs (first time mentioning CLI)

Reply:
You're right. I have to extend that section. Will do that early next week.